src/js/view/floating/floating-controller.ts

```diff
@@ -0,0 +1,371 @@
+import type Model from 'controller/model';
+import type Preview from 'view/preview';
+import type { FloatConfig } from 'controller/model';
+import FloatingDragUI from 'view/floating/floating-drag-ui';
+import { OS } from 'environment/environment';
+import { deviceIsLandscape } from 'utils/dom';
+import { isIframe } from 'utils/browser';
+import {
+    addClass,
+    removeClass
+} from 'utils/dom';
+import { STATE_IDLE, STATE_COMPLETE, STATE_ERROR } from 'events/events';
+import { isNumber } from 'utils/underscore';
+import {
+    style,
+} from 'utils/css';
+import viewsManager from 'view/utils/views-manager';
+import type { BoundsObj } from 'types/generic.type';
+import { getPlayerSizeStyles } from 'view/utils/player-size';
+
+const FLOATING_TOP_OFFSET = 62;
+
+let _floatingPlayer: HTMLElement | null = null;
+const fpContainer: {
+    floatingPlayer: HTMLElement | null;
+} = {
+    floatingPlayer: _floatingPlayer
+};
+Object.defineProperty(fpContainer, 'floatingPlayer', {
+    get: () => _floatingPlayer,
+    set: (val) => {
+        if (val === _floatingPlayer) {
+            return;
+        }
+        _floatingPlayer = val;
+        const watchersToRun = watchers.concat([]);
```

Comment:
Copy arrays without the use of an array literal:
```suggestion
        const watchersToRun = watchers.concat([]);
        const watchersToRun = watchers.slice();
```

Reply:
Great suggestion :+1: